contracts/utils/Address.sol

```diff
@@ -49,21 +79,25 @@ library Address {
      * _Available since v3.1._
      */
     function functionCall(address target, bytes memory data) internal returns (bytes memory) {
-        return functionCallWithValue(target, data, 0, "Address: low-level call failed");
+        return functionCallWithValue(target, data, 0, defaultOnRevert);
     }
 
     /**
-     * @dev Same as {xref-Address-functionCall-address-bytes-}[`functionCall`], but with
-     * `errorMessage` as a fallback revert reason when `target` reverts.
+     * @dev Same as {xref-Address-functionCall-address-bytes-}[`functionCall`], but with an
+     * `onRevert` function as a fallback when `target` reverts.
+     *
+     * Requirements:
+     *
+     * - `onRevert` must be a reverting function.
      *
      * _Available since v3.1._
      */
     function functionCall(
         address target,
         bytes memory data,
-        string memory errorMessage
+        function() internal pure onRevert
```

Comment:
Maybe pure is too restrictive. I'd allow view functions.
(comment apply to all function that take an onRevert fnpointer)

Reply:
I went for pure since I don't completely know the security implications of making it view. I think it requires a bit more research since `view` might have consequences. I'm thinking about something similar to a view-only reentrancy but not sure if it's actually a problem.

Question for you and @frangio, do you see any security implication of making it `view` instead? If not, I'll proceed to make the change, but please confirm.

Reply:
I don't see any security implications, I would make it view so that we don't limit unnecessarily.

Reply:
My point for using view is that you may want the error to return some value. You can do that by storing the value (tstore?), then reading it inside the `onRevert`. But that requires the `onRevert` to be able to read storage.

Reply:
@Amxx Yes, I see the value of using `view` to include parameters.

Just added, thanks both for your comments!

Reply:
After we call the `onRevert` function, we force a revert (just in case the function does not throw). I think this is great for security. This also means that whatever `onRevert` does WILL be reverted. I believe this means there is no way `onRevert` does any attack. 
I believe it would be safe to remove the view, but I also don't see the point to doing it. View should be enough.